### PR TITLE
Parse known Gutenberg blocks and extract strings for translation

### PR DESCRIPTION
**Pull request by brucepearson**
Parse known Gutenberg blocks and extract strings for translation - https://onthegosystems.myjetbrains.com/youtrack/issue/wpmlcore-5653

### CODE REVIEW

**Comment by brucepearson** [comment]
@OnTheGoSystems/wpml Please review

File: src/class-wpml-gutenberg-strings-in-block.php [hunk 1]
```diff
@@ -0,0 +1,213 @@
+<?php
+
+/**
+ * Class WPML_Gutenberg_Strings_In_Block
+ */
+class WPML_Gutenberg_Strings_In_Block {
+
+	/** @var array */
+	private $block_types = array(
```
Comment (brucepearson) [inline src/class-wpml-gutenberg-strings-in-block.php]: **Note:** These are xpath queries to find the element or attribute that has text needing translation. These will come from `wpml-config.xml` and will be handled in the following merge request.

File: src/class-wpml-gutenberg-strings-in-block.php [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+		'core/paragraph' => array( '//p' ),
+		'core/heading'   => array( "//*[name()='h1' or name()='h2' or name()='h3' or name()='h4' or name()='h5' or name()='h6']" ),
```
Comment (andreasciamanna) [inline src/class-wpml-gutenberg-strings-in-block.php]: @BruceOnTheGo I think that's a better way to do the same: `//*[self:: h1 or self:: h2 or self::h3 or self::h4 or self::h5 or self::h6]`.

It also looks like your solution may have unpredictable effects in some circumstances: https://stackoverflow.com/a/722251/626538

**Review by andreasciamanna — APPROVED** [review]
It looks ok. I'm not 100% sure about the xPath query used for `h*` elements, though.

File: src/class-wpml-gutenberg-strings-in-block.php (hunk 2 above)
Comment (brucepearson) [reply src/class-wpml-gutenberg-strings-in-block.php]: @andreasciamanna I've updated the query in the wpml-config.xml file - https://git.onthegosystems.com/wpml/sitepress-multilingual-cms/merge_requests/1986/diffs

**Pull request merged**